pkg/util/intstr/intstr.go

```diff
@@ -67,7 +71,17 @@ func (intstr *IntOrString) String() string {
 	if intstr.Type == String {
 		return intstr.StrVal
 	}
-	return strconv.Itoa(intstr.IntVal)
+	return strconv.Itoa(intstr.IntValue())
+}
+
+// IntValue returns the IntVal if type Int, or if
+// it is a String, will attempt a conversion to int.
+func (intstr *IntOrString) IntValue() int {
+	if intstr.Type == String {
```

Comment:
It was to prevent having to refactor 900 lines of internal code that assume
IntOrString is int, not int32.  The string conversion bit can be removed as
long as we doc it (all the places it was used was the IntValue)

On Mon, Nov 30, 2015 at 1:27 PM, Tim Hockin <notifications@github.com>
wrote:

> In pkg/util/intstr/intstr.go
> <https://github.com/kubernetes/kubernetes/pull/17353#discussion_r46181719>
> :
>
> > @@ -67,7 +71,17 @@ func (intstr *IntOrString) String() string {
> >  	if intstr.Type == String {
> >  		return intstr.StrVal
> >  	}
> > -	return strconv.Itoa(intstr.IntVal)
> > +	return strconv.Itoa(intstr.IntValue())
> > +}
> > +
> > +// IntValue returns the IntVal if type Int, or if
> > +// it is a String, will attempt a conversion to int.
> > +func (intstr *IntOrString) IntValue() int {
> > +	if intstr.Type == String {
>
> I think this is wrong and we should not do it. It's scope creep for this
> PR anyway. Was there a use case justifying this? I think we should roll it
> back
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/17353/files#r46181719>.
>